BasicHistorySessionsKotlin/app/src/main/java/com/google/android/gms/fit/samples/basichistorysessionskotlin/MainActivity.kt

```diff
@@ -0,0 +1,513 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.gms.fit.samples.basichistorysessionskotlin
+
+import android.Manifest
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.graphics.Color
+import android.net.Uri
+import android.os.Bundle
+import android.provider.Settings
+import android.view.Menu
+import android.view.MenuItem
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.widget.TextViewCompat
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.fit.samples.common.logger.Log
+import com.google.android.gms.fit.samples.common.logger.LogView
+import com.google.android.gms.fit.samples.common.logger.LogWrapper
+import com.google.android.gms.fit.samples.common.logger.MessageOnlyLogFilter
+import com.google.android.gms.fitness.Fitness
+import com.google.android.gms.fitness.FitnessActivities
+import com.google.android.gms.fitness.FitnessOptions
+import com.google.android.gms.fitness.data.DataPoint
+import com.google.android.gms.fitness.data.DataSet
+import com.google.android.gms.fitness.data.DataSource
+import com.google.android.gms.fitness.data.DataType
+import com.google.android.gms.fitness.data.Field
+import com.google.android.gms.fitness.data.Session
+import com.google.android.gms.fitness.request.DataDeleteRequest
+import com.google.android.gms.fitness.request.SessionInsertRequest
+import com.google.android.gms.fitness.request.SessionReadRequest
+import com.google.android.gms.fitness.result.SessionReadResponse
+import com.google.android.gms.tasks.Task
+import com.google.android.material.snackbar.Snackbar
+import java.util.Calendar
+import java.util.Date
+import java.util.TimeZone
+import java.util.concurrent.TimeUnit
+
+const val TAG = "BasicSessions"
+const val SAMPLE_SESSION_NAME = "Afternoon run"
+
+/**
+ * This sample demonstrates how to use the Sessions API of the Google Fit platform to insert
```

Comment:
I'm not a fitness expert, but how are sessions different than putting data in via the history API? Might be something obvious I don't see. ^_^;;

Either way, might be nice to call out why you would use it for noobs like me.

Reply:
Added